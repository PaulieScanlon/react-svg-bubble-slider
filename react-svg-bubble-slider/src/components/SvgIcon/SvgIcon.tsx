import React, { FunctionComponent, HTMLAttributes } from 'react'

import { IconSetOptions, IconSetProps } from '../types'
import { iconPaths } from '../SvgBubbleSlider/iconPaths'

import { createIconPathsMarkup } from '../utils'

import theme from '../../theme'

interface SvgIconProps extends IconSetProps, HTMLAttributes<SVGElement> {
  /** The name of icon */
  name: string
  /** The width and height of the icon */
  size?: number
  /** The fill color */
  color?: string
}

export const SvgIcon: FunctionComponent<SvgIconProps> = ({
  name,
  size = 32,
  color = theme.colors.primary,
  iconSet = IconSetOptions.chrisGannon,
}: SvgIconProps) => {
  if (size < 32) throw new Error('Icon size must be at least 32')
  return (
    <svg
      className="svg-icon"
      xmlns="http://www.w3.org/2000/svg"
      width={size}
      height={size}
      color={color}
      fill="currentcolor"
      viewBox="0 0 32 32"
      preserveAspectRatio="xMidYMid meet"
      x="0"
      y="0"
    >
      {iconPaths[iconSet]
        .filter((icon) => icon.name === name)
        .map((icon, index) => {
          const { paths } = icon
          return (
            <g
              // transform={`matrix(0.8,0,0,0.8,0,0)`}
              key={index}
              dangerouslySetInnerHTML={createIconPathsMarkup(paths)}
              fill="currentcolor"
            />
          )
        })}
    </svg>
  )
}

SvgIcon.displayName = 'SvgIcon'
