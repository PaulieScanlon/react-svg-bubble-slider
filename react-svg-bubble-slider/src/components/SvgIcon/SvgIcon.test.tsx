import React from 'react'
import { render } from '@testing-library/react'

import { SvgIcon } from './SvgIcon'

describe('<SvgIcon />', () => {
  test('renders the component', () => {
    const { asFragment } = render(<SvgIcon name="happy" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
