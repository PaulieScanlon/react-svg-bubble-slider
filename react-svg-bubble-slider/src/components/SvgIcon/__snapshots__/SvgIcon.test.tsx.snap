// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SvgIcon /> renders the component 1`] = `
<DocumentFragment>
  <svg
    class="svg-icon"
    color="#FF69B4"
    fill="currentcolor"
    height="32"
    preserveAspectRatio="xMidYMid meet"
    viewBox="0 0 32 32"
    width="32"
    x="0"
    xmlns="http://www.w3.org/2000/svg"
    y="0"
  >
    <g
      fill="currentcolor"
    >
      <path
        d="M16,0C7.2,0,0,7.2,0,16s7.2,16,16,16s16-7.2,16-16S24.8,0,16,0z M22,8c1.1,0,2,1.3,2,3s-0.9,3-2,3s-2-1.3-2-3S20.9,8,22,8z M10,8c1.1,0,2,1.3,2,3s-0.9,3-2,3s-2-1.3-2-3S8.9,8,10,8z M16,28c-5.2,0-9.5-4.4-10-9.9c2.9,1.7,6.4,2.7,10,2.7s7.1-1,10-2.7C25.5,23.6,21.2,28,16,28L16,28z"
      />
    </g>
  </svg>
</DocumentFragment>
`;
